Source/Scene/ShadowMapShader.js

```diff
@@ -0,0 +1,368 @@
+/*global define*/
+define([
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Renderer/ShaderSource'
+    ], function(
+        defaultValue,
+        defined,
+        ShaderSource) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    function ShadowMapShader() {
+    }
+
+    function findVarying(shaderSource, names) {
```

Comment:
For helpers, I just mean `findNormalVarying` and `findPositionVarying`, not the shadow-map-specific functions, which are well placed here.